taca/illumina/HiSeq_Runs.py

```diff
@@ -92,138 +92,155 @@ def compute_undetermined(self):
         NoIndexLanes = [lane["Lane"] for lane in self.runParserObj.samplesheet.data if "NoIndex" in lane["index"]]
         if len(NoIndexLanes) == 0:
             return True # everything is fine I can proceed to QC
-        else:
-            NoIndex_Undetermiend = os.path.join(self.run_dir, "Demultiplexing_NoIndex")
-            if not os.path.exists(NoIndex_Undetermiend):
-                os.makedirs(NoIndex_Undetermiend)
-                #for these lanes I have no undetermiend as I demux them without index.
-                #now geenrate the base masks per lane
-                per_lane_base_masks = self._generate_per_lane_base_mask()
-                #store here only the NoIndex lanes
-                per_lane_base_masks_NoIndex = {}
-                for NoIndexLane in NoIndexLanes:
-                    per_lane_base_masks_NoIndex[NoIndexLane] = per_lane_base_masks[NoIndexLane]
-                    base_mask_key = per_lane_base_masks[NoIndexLane].keys()[0]
-                    new_base_mask = []
+        #otherwise proceed
+        
+        NoIndex_Undetermiend = os.path.join(self.run_dir, "Demultiplexing_NoIndex")
+        if not os.path.exists(NoIndex_Undetermiend):
+            #for these lanes I have no undetermiend as I demux them without index.
+            #now geenrate the base masks per lane
+            per_lane_base_masks = self._generate_per_lane_base_mask()
+            #store here only the NoIndex lanes
+            per_lane_base_masks_NoIndex = {}
+            run_with_no_index = False # use this flag to check that we are not in the C.Daub case
+            for NoIndexLane in NoIndexLanes:
+                per_lane_base_masks_NoIndex[NoIndexLane] = per_lane_base_masks[NoIndexLane]
+                base_mask_key = per_lane_base_masks[NoIndexLane].keys()[0]
+                new_base_mask = []
+                if len(per_lane_base_masks_NoIndex[NoIndexLane][base_mask_key]['base_mask']):
```

Comment:
yeah, you don't need to use len for arrays either, [] is evaluated to False
and [<any number of anything>] is evaluated to true. Doesnt change much.

On Wed, Oct 28, 2015 at 4:25 PM, Francesco <notifications@github.com> wrote:

> In taca/illumina/HiSeq_Runs.py
> <https://github.com/SciLifeLab/TACA/pull/143#discussion_r43270373>:
>
> > -                    new_base_mask = []
> > +        #otherwise proceed
> > +
> > +        NoIndex_Undetermiend = os.path.join(self.run_dir, "Demultiplexing_NoIndex")
> > +        if not os.path.exists(NoIndex_Undetermiend):
> > +            #for these lanes I have no undetermiend as I demux them without index.
> > +            #now geenrate the base masks per lane
> > +            per_lane_base_masks = self._generate_per_lane_base_mask()
> > +            #store here only the NoIndex lanes
> > +            per_lane_base_masks_NoIndex = {}
> > +            run_with_no_index = False # use this flag to check that we are not in the C.Daub case
> > +            for NoIndexLane in NoIndexLanes:
> > +                per_lane_base_masks_NoIndex[NoIndexLane] = per_lane_base_masks[NoIndexLane]
> > +                base_mask_key = per_lane_base_masks[NoIndexLane].keys()[0]
> > +                new_base_mask = []
> > +                if len(per_lane_base_masks_NoIndex[NoIndexLane][base_mask_key]['base_mask']):
>
> per_lane_base_masks_NoIndex[NoIndexLane][base_mask_key]['base_mask'] is
> an array, so I check for the length of an array, so is greater than 0
> evaluates to True
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/TACA/pull/143/files#r43270373>.
>